src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +111,624 @@ static InitChainEntry D_80936754[] = {
+        this->actor.floorHeight =
+            BgCheck_EntityRaycastFloor5(&globalCtx->colCtx, &this->actor.floorPoly, &bgId, &this->actor, &pos);
+    }
+}
+
+void EnKusa_SetupFall(EnKusa* this) {
+    this->actionFunc = EnKusa_Fall;
+    rotSpeedXtarget = -0xBB8;
+    rotSpeedYtarget = (Rand_ZeroOne() - 0.5f) * 1600.0f;
+    rotSpeedX = 0;
+    rotSpeedY = 0;
+    this->timer = 0;
+}
+
+void EnKusa_Fall(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 wasHit;
+    Vec3f contactPos;
+    s32 i;
+    s16 angleOffset;
+
+    wasHit = (this->collider.base.atFlags & AT_HIT) != 0;
+    if (wasHit) {
+        this->collider.base.atFlags &= ~AT_HIT;
+    }
+    this->timer++;
+    if ((this->actor.bgCheckFlags & 0xB) || (wasHit) || (this->timer >= 100)) {
+        if (!(this->actor.bgCheckFlags & 0x20)) {
+            SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_PLANT_BROKEN);
+        }
+        EnKusa_SpawnFragments(this, globalCtx);
+        EnKusa_DropCollectible(this, globalCtx);
+        switch (GET_KUSA_TYPE(&this->actor)) {
+            case ENKUSA_TYPE_BUSH:
+            case ENKUSA_TYPE_GRASS:
+                Actor_MarkForDeath(&this->actor);
+                break;
+
+            case ENKUSA_TYPE_REGROW_GRASS:
+                EnKusa_SetupUprootedWaitRegrow(this);
+                this->actor.shape.shadowDraw = NULL;
+                break;
+        }
+
+    } else {
+        if (this->actor.bgCheckFlags & 0x40) {
+            contactPos.y = this->actor.world.pos.y + this->actor.depthInWater;
+            for (angleOffset = 0, i = 0; i < 4; i++, angleOffset += 0x4000) {
+                contactPos.x =
+                    (Math_SinS((s32)(Rand_ZeroOne() * 7200.0f) + angleOffset) * 15.0f) + this->actor.world.pos.x;
+                contactPos.z =
+                    (Math_CosS((s32)(Rand_ZeroOne() * 7200.0f) + angleOffset) * 15.0f) + this->actor.world.pos.z;
+                EffectSsGSplash_Spawn(globalCtx, &contactPos, NULL, NULL, 0, 190);
+            }
+            contactPos.x = this->actor.world.pos.x;
+            contactPos.z = this->actor.world.pos.z;
+            EffectSsGSplash_Spawn(globalCtx, &contactPos, NULL, NULL, 0, 280);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 300, 700, 0);
+            this->actor.terminalVelocity = -3.0f;
+            this->actor.velocity.x *= 0.1f;
+            this->actor.velocity.y *= 0.4f;
+            this->actor.velocity.z *= 0.1f;
+            this->actor.gravity *= 0.5f;
+            rotSpeedX >>= 1;
+            rotSpeedXtarget >>= 1;
+            rotSpeedY >>= 1;
+            rotSpeedYtarget >>= 1;
+            this->actor.bgCheckFlags &= ~0x40;
+            SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 40, NA_SE_EV_DIVE_INTO_WATER_L);
+        }
+        EnKusa_UpdateVelY(this);
+        Math_StepToS(&rotSpeedX, rotSpeedXtarget, 0x1F4);
+        Math_StepToS(&rotSpeedY, rotSpeedYtarget, 0xAA);
```

Comment:
I think speeds in dec. Also, do they have to be shifts above?

Reply:
Yes the shifts are needed